database-commons/src/main/java/io/cdap/plugin/db/DBRecord.java

```
@@ -212,13 +244,67 @@ public void write(DataOutput out) throws IOException {
+    operationName = operationName == null ? Operation.INSERT :
+      Operation.valueOf(operationName.toString().toUpperCase());
+    switch (operationName) {
+      case INSERT:
+        insertOperation(stmt);
+        break;
+      case UPDATE:
+        updateOperation(stmt);
+        break;
+      case UPSERT:
+        upsertOperation(stmt);
+        break;
+    }
+  }
+
+  protected void insertOperation(PreparedStatement stmt) throws SQLException {
     for (int fieldIndex = 0; fieldIndex < columnTypes.size(); fieldIndex++) {
       ColumnType columnType = columnTypes.get(fieldIndex);
       Schema.Field field = record.getSchema().getField(columnType.getName());
       writeToDB(stmt, field, fieldIndex);
     }
   }
 
+  protected void updateOperation(PreparedStatement stmt) throws SQLException {
+    List<String> updatedKeyList = Arrays.asList(relationTableKey.split(","));
+    for (int fieldIndex = 0; fieldIndex < columnTypes.size(); fieldIndex++) {
+      ColumnType columnType = columnTypes.get(fieldIndex);
+      Schema.Field field = record.getSchema().getField(columnType.getName());
+      writeToDB(stmt, field, fieldIndex);
+      if (fillUpdateParams(updatedKeyList, columnType)) {
+        modifiableColumnTypes.add(columnType);
+      }
+    }
+
+    // Used for filling the question marks for update
+    if (operationName != null && relationTableKey != null) {
+      for (int fieldIndex = columnTypes.size(); fieldIndex < modifiableColumnTypes.size(); fieldIndex++) {
+        ColumnType columnType = modifiableColumnTypes.get(fieldIndex);
+        Schema.Field field = record.getSchema().getField(columnType.getName());
+        writeToDB(stmt, field, fieldIndex);
+      }
+    }
+  }
+
+  /**
+   * Upsert is different for all plugins. So, will be overriding this method to write to query.
+   * @param stmt
+   * @throws SQLException
+   */
+  protected void upsertOperation(PreparedStatement stmt) throws SQLException {
+    return;
```

Throw UnsupportedOperationException instead of simple return

---

resolved